### DISCUSSION TITLE
Nested query, and, many or

### DISCUSSION THREAD

**Discussion by damiankomonski** [prisma/prisma#22332] [Q&A]
### Question

and I have question, how can I write query when I have 3 tables, one is posts, second users, third payments. I want get all posts where users name is "abc" or "bca" and payments id is "abc" or "abc2". How can I write such method?

I tried this but it doesnt work

```
prisma.posts.findMany({
    AND: [
        {
            OR: [
               {
                  users: {is: {name: "abc"}}
               },
                {
                  users: {is: {name: "bca"}}
               },
          ]
        },
        {
            OR: [
               {
                  payments: {is: {name: "abc"}}
               },
                {
                  payments: {is: {name: "abc2"}}
               },
          ]
        }
    ]
})
```

It doesnt work... help please.

### Information about Prisma Schema, Client Queries and Environment (optional)

```prisma
// Add your schema.prisma
```

```ts
// Add any relevant Prisma Client queries here
```

- OS: 
- Database: 
- Node.js version: 

- Run `prisma -v` to see your Prisma version and paste it

**Comment by damiankomonski** [comment]
I need something like that in prisma 

```
SELECT
  student.first_name,
  student.last_name,
  course.name
FROM student
JOIN student_course
  ON student.id = student_course.student_id
JOIN course
  ON course.id = student_course.course_id;
WHERE (student_course.name="Abc" OR student_course.name="abc2") AND (course.name="asd" OR course.name="asd12")
```

**Comment by ludralph** [comment]
Hi @damiankomonski 👋 

Thank you for raising this question.

Can you please share a minimal schema that contains the `Posts`, `Users` and `Payments` models and the relationships between them?

**Comment by MartinRitch** [comment]
Hi,
I've a similar request. A table user, another one contact and a third user2contact.

model User {
  id           Int            @id @default(autoincrement())
  guid         String         @unique @default(uuid())
  email        String         @unique
  username     String         @unique
  preventDel   Boolean        @default(true)
  isEnabled    Boolean        @default(false)
  isConfirmed  Boolean        @default(false)
  User2Contact User2Contact[] @relation("User")
  Role2User    Role2User[]    @relation("RoleUser")

  @@map("user")
}

model Contact {
  id                    Int                     @id @default(autoincrement())
  guid                  String                  @unique @default(uuid())
  firstname             String
  lastname              String
  nickname              String?
  street                String?
  zip                   String?
  city                  String?
  mobile                String?
  email                 String?
  Contact2Client        Contact2Client[]        @relation("ContactClient")
  User2Contact          User2Contact[]          @relation("UserContact")

  @@map("contact")
}

model User2Contact {
  id          Int      @id @default(autoincrement())
  contactGuid String   @unique
  Contact     Contact  @relation("UserContact", fields: [contactGuid], references: [guid])
  userGuid    String   @unique
  User        User     @relation("User", fields: [userGuid], references: [guid])

  @@map("user2contact")
}

and I need to do a query like this

select u.guid ,
	u.email ,
	u.username ,
	uc."userGuid" ,
	uc."contactGuid" ,
	c.guid ,
	c.firstname ,
	c.lastname ,
	c.nickname
from "user" u 
left join user2contact uc on u.guid = uc."userGuid" 
left join contact c on uc."contactGuid" = c.guid ;


Is there a way to do this direct in prisma directly or do I need a raw query for that? And if I need a raw can I do this with a view or better with a raw query?

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
